src/main/java/edu/ie3/datamodel/io/sink/SqlSink.java

```diff
@@ -0,0 +1,521 @@
+        insert(entity, identifier);
+    }
+
+    // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+
+    protected <C extends UniqueEntity> void persistListIncludeNested(List<C> entities, Class<C> cls, DatabaseIdentifier identifier) throws SQLException {
+        if (NestedEntity.class.isAssignableFrom(cls)) {
+            entities.forEach(
+                    entity -> {
+                        try {
+                            List<InputEntity> arr = new ArrayList<>(Extractor.extractElements((NestedEntity) entity));
+                            persistAll(arr, identifier);
+                        } catch (ExtractorException | SQLException e) {
+                            log.error(
+                                    String.format(
+                                            "An error occurred during extraction of nested entity'%s': ",
+                                            cls.getSimpleName()),
+                                    e);
+                        }
+                    }
+            );
+            insertListIgnoreNested(entities, cls, identifier);
+        } else {
+            insertListIgnoreNested(entities, cls, identifier);
+        }
+    }
+
+    public <C extends UniqueEntity> void persistIncludeNested(C entity, DatabaseIdentifier identifier) throws SQLException {
+        Set<C> entitiesToAdd = new HashSet<>();
+        entitiesToAdd.add(entity);
+        persistAll(entitiesToAdd, identifier);
+    }
+
+    private <C extends UniqueEntity> void persistMixedList(List<C> entities, DatabaseIdentifier identifier) {
+        Map<Class<C>, List<C>> entitiesPerClass = entities.stream()
+                .collect(groupingBy(entity -> (Class<C>) entity.getClass()));
+        entitiesPerClass.forEach(
+                (cls, ent) -> {
+                    try {
+                        persistList(ent, cls, identifier);
+                    } catch (SQLException e) {
+                        log.error(String.format(
+                                        "An error occurred during extraction of entity'%s': ",
+                                        cls.getSimpleName()),
+                                e);
+                        throw new RuntimeException(String.format(
+                                "An error occurred during extraction of entity '%s', SQLReason: '%s'",
+                                cls.getSimpleName(), e.getMessage()), e);
+                    }
+                }
+        );
+    }
+
+    private <C extends UniqueEntity, E extends TimeSeriesEntry<V>, V extends Value> void persistList(List<C> entities, Class<C> cls, DatabaseIdentifier identifier) throws SQLException {
+        // Check if there are only elements of the same class
+        Class<?> firstClass = entities.get(0).getClass();
+        boolean allSameClass = true;
```

Comment:
Not important but a more concise way to write that could be: 

```suggestion
         boolean allSameClass = entities.stream().allMatch(e -> e.getClass() == firstClass);
```